prototype/gmail_accounts.py

```diff
@@ -0,0 +1,101 @@
+import string
+from httplib2 import Http
+from random import choice
+
+from apiclient.errors import HttpError
+from apiclient.discovery import build
+from oauth2client.service_account import ServiceAccountCredentials
+
+
+PASSWORD_CHARS = string.ascii_letters + string.digits
+
+def generate_password(length=10):
+    return ''.join([choice(PASSWORD_CHARS) for i in range(length)])
+
+
+
+"""
+Several things were needed to get this working:
+1. Create an app in Developer Console
+2. Create a service account to enable 2 legged oauth (https://developers.google.com/identity/protocols/OAuth2ServiceAccount)
+3. Enable delegation of domain-wide authority for the service account.
+4. Enable Admin SDK for the domain.
+5. Give the service account permission to access admin.directory.users service (https://admin.google.com/AdminHome?chromeless=1#OGX:ManageOauthClients).
+"""
+
+SCOPES = 'https://www.googleapis.com/auth/admin.directory.user'
+
+credentials = ServiceAccountCredentials.from_json_keyfile_name(
+    'djangogirls_website-93134565f52d.json',
+    scopes=SCOPES
+)
+
+delegated_credentials = credentials.create_delegated('hello@djangogirls.org')
+http_auth = delegated_credentials.authorize(Http())
+
+service = build('admin', 'directory_v1', http=http_auth)
+
+
+def make_email(slug):
+    """Get the email address for the given slug"""
+    return '%s@djangogirls.org' % slug
+
+
+def create_account(slug, city_name):
+    """
+    Create a new account
+
+    e.g. create_account('testcity', 'Test City')
+    """
+    email = make_email(slug)
+    password = generate_password()
+    response = service.users().insert(body={
```

Comment:
It'll raise an HttpError (with code 409). I'll document that in the comment for now.